### PR TITLE
feat: gw_mainnet_v1 regression testing workflow

### PR DESCRIPTION
**Pull request by ShookLyngs**
### Description
The PR provides a mainnet_v1 regression testing workflow that can be triggered manually on GitHub.

### Provide changes:
- Add a regression testing workflow
- Add root hooks to check total capacity spent of the run (for each account)
- Add util functions to check networks, most test cases are disabled when running `gw_mainnet_v1`

### Supported test cases
- TestStatus
- BlockInfo
- sUDT-ERC20-Proxy

### New secrets
The PR provides a new GitHub workflow, which requires some new secrets:
- GW_MAINNET_V1_PK1
- GW_MAINNET_V1_PK2

### Network identity
We used to identify current networking by checking network's `chainId`, but I think it's better to use `enum` and `network.name`. 

Based on that idea, the PR provides a set of util functions (and enum values) to check the identity of current network by its `name`.

The new set of util functions is not perfect, if you think there's better way to do it, please let me know. 

Right now, this is how we check the identity of current networking:
```javascript
// check with predefined enum
if (isNetwork(Networks.GwMainnet_v1)) {
  return;
}

// check with specific check-function
if (isGwMainnetV1()) {
  return;
}
```

### Total spent of the run
The PR provides a new test case to replace the `network` test case, which will print logs when all tests are done:

```
Total capacity changed in the test run:
0x966B30e576A4d6731996748B48Dd67C94eF29067 spent -1115.871750000000013888 pCKB (-1115871750000000013888 capacity)
0x4fEF21F1d42E0D23d72100aEFe84d555781c31BB spent -15.49854 pCKB (-15498540000000000000 capacity)
0x0C1EfCCa2Bcb65A532274f3eF24c044EF4ab6D73 spent +0.0 pCKB (0 capacity)
```

The logs above give us a clear view of how much capacity is spent in current run. 

Although it may not perfectly reflect the actual cost in every situation (if we run the same test with the same set of accounts at the same time, the calculated info must be wrong), but I think logging it is better than doing nothing.

### CODE REVIEW

File: contracts/utils/network.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const { network } = require("hardhat");
+
+const Networks = {
+  GwDevnet_v1: 'gw_devnet_v1',
+  GwTestnet_v1: 'gw_testnet_v1',
+  GwAlphanet_v1: 'gw_alphanet_v1',
+  GwBetanet_v1: 'gw_betanet_v1',
```
Comment (Flouse) [inline contracts/utils/network.js]: I think we could deprecate gw_betanet_v1 now.

File: contracts/utils/network.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+function isGwMainnetV1() {
+  return isNetwork(Networks.GwMainnet_v1);
+}
+
+module.exports = {
+  Networks,
+  isNetwork,
+  isGwMainnetV1,
+};
```
Comment (Flouse) [inline contracts/utils/network.js]:
```suggestion
};

```

File: contracts/test/sUDT-ERC20-Proxy.js [hunk 3]
```diff
@@ -11,134 +11,160 @@
...
-  it("Transfer GWK", async function() {
-    // should be an EOA containing some GWK
-    const sender = (await ethers.getSigners())[2];
+  it(`Transfer ${transferSudt.name}`, async () => {
+    const [sender, receiver] = await ethers.getSigners();
+    const sudt = await ethers.getContractAt(abi, transferSudt.address);
+
+    const [senderBalance, receiverBalance] = await Promise.all([
+      await sudt.callStatic.balanceOf(sender.address, { gasPrice: 0 }),
+      await sudt.callStatic.balanceOf(receiver.address, { gasPrice: 0 }),
```
Comment (Flouse) [inline contracts/test/sUDT-ERC20-Proxy.js]:
```suggestion
      sudt.callStatic.balanceOf(sender.address, { gasPrice: 0 }),
      sudt.callStatic.balanceOf(receiver.address, { gasPrice: 0 }),
```

**Review by Flouse — CHANGES_REQUESTED** [review]

File: contracts/utils/network.js (hunk 1 above)
Comment (ShookLyngs) [reply contracts/utils/network.js]: Okay, do you think we should use `isGwMainnetV1()` or use `isNetwork(Networks.GwMainnet_v1)` directly?
Or should we have functions like:
```
runIf(isNetwork(Networks.GwMainnet_v1), () => {
  describe(...);
});
```

Would it be matter?

Comment (Flouse) [reply contracts/utils/network.js]: `isGwMainnetV1()` LGTM.

**Review by Flouse — APPROVED** [review]

**Pull request merged**